Por ello, conviene extraer el cuerpo de los bloques  try  y  catch  en funciones individuales. public void delete(Page page) { try { deletePageAndAllReferences(page); } catch (Exception e) { logError(e); } } private void deletePageAndAllReferences(Page page) throws Exception { deletePage(page); registry.deleteReference(page.name); configKeys.deleteKey(page.name.makeKey()); } private void logError(Exception e) { logger.log(e.getMessage()); } En este caso, la función  delete  es de procesamiento de errores. Es fácil de entender e ignorar. La función  deletePageAndAllReferences  es para los procesos de borrar una página. El procesamiento de errores se puede ignorar. De este modo, la separación facilita la comprensión y la modificación del código. El procesamiento de errores es una cosa Las funciones sólo deben hacer una cosa y el procesamiento de errores es un ejemplo. Por tanto, una función que procese errores no debe hacer nada más. Esto implica (como en el ejemplo anterior) que, si una función incluye la palabra clave  try , debe ser la primera de la función y que no debe haber nada más después de los bloques  catch/finally . El imán de dependencias Error.java La devolución de códigos de error suele implicar que existe una clase o enumeración en la que se definen los códigos de error. public enum Error { 80